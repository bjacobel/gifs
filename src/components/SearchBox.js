import React, { Component } from 'react';
import { connect } from 'react-redux';

import { searchGifsAsync } from '../actions/search';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  searchGifsAsync,
};

class SearchBox extends Component {
  search(event) {
    this.props.searchGifsAsync(event.currentTarget.value);
  }

  render() {
    return (
      <div className="search-box">
        <input
          placeholder="Search"
          type="search"
          onChange={ event => this.search(event) }
        />
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchBox);
