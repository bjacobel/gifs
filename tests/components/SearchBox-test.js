import React from 'react';
import { shallow, mount } from 'enzyme';

import SearchBox from '../../src/components/SearchBox';

describe('SearchBox component', () => {
  it('matches snapshot', () => {
    expect(shallow(<SearchBox.WrappedComponent searchGifsAsync={ jest.fn() } />)).toMatchSnapshot();
  });

  it('matches snapshot when it gets an initialSearch param', () => {
    expect(
      mount( // must use mount so ref callback is run
        <SearchBox.WrappedComponent
          searchGifsAsync={ jest.fn() }
          searchIndex={ ['idk'] }
          initialSearch="foo"
        />,
      ),
    ).toMatchSnapshot();
  });

  it('sets the value through use of the ref callback', () => {
    expect(
      mount( // must use mount so ref callback is run
        <SearchBox.WrappedComponent
          searchGifsAsync={ jest.fn() }
          searchIndex={ ['idk'] }
          initialSearch="foo"
        />,
      ).find('input').node.value,
    ).toEqual('foo');
  });

  it('calls searchGifsAsync when input changes', () => {
    const searchGifsAsync = jest.fn();
    const enzymeRepr = shallow(<SearchBox.WrappedComponent searchGifsAsync={ searchGifsAsync } />);
    enzymeRepr.find('input')
      .simulate('change', { currentTarget: { value: 'a' } })
      .simulate('change', { currentTarget: { value: 'b' } })
      .simulate('change', { currentTarget: { value: 'c' } });
    expect(searchGifsAsync).toHaveBeenCalledTimes(3);
  });

  it('calls searchGifsAsync with initialSearch if passed it', () => {
    const searchGifsAsync = jest.fn();
    shallow(
      <SearchBox.WrappedComponent
        searchGifsAsync={ searchGifsAsync }
        initialSearch="foo"
        searchIndex={ ['idk'] }
      />,
    );
    expect(searchGifsAsync).lastCalledWith('foo');
  });
});
