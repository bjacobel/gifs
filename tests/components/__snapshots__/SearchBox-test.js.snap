exports[`SearchBox component matches snapshot 1`] = `
<div
  className="search-box">
  <input
    onChange={[Function]}
    placeholder="Search"
    type="search" />
</div>
`;
